src/main/scala/org/constellation/API.scala

```diff
@@ -229,14 +255,33 @@ class API(udpAddress: InetSocketAddress)(implicit system: ActorSystem,
     post {
       pathPrefix("channel") {
         path("open") {
-          entity(as[ChannelOpenRequest]) { request =>
-            ChannelMessage.createGenesis(request)
-            complete(StatusCodes.Created)
+          entity(as[ChannelOpen]) { request =>
+            onComplete(
+              ChannelMessage.createGenesis(request)
+            ) { res =>
+              complete(res.getOrElse(ChannelOpenResponse("API Failure")))
+            }
+
           }
         } ~
           path("send") {
             entity(as[ChannelSendRequest]) { send =>
-              complete(ChannelMessage.createMessages(send))
+              onComplete(ChannelMessage.createMessages(send)) { res: Try[ChannelSendResponse] =>
+                complete(res.getOrElse(ChannelOpenResponse("API Failure")).json)
```

Comment:
If this is chanel send request why are you using `ChannelOpenResponse` instead of `ChannelSendResponse` IMO it's a bit misleading, and message `API Failure` actually doesn't mean anything

Reply:
Absolutely, this should be a ChannelSendResponse

Reply:
I've updated the messages in the case of failures for each as well